src/Tvheadend.cpp

```diff
@@ -1645,7 +1663,12 @@ void* CTvheadend::Process ( void )
   {
     /* Check Q */
     // this is a bit horrible, but meh
-    if (!m_queue.Pop(msg, 2000))
+    bool bSuccess = m_queue.Pop(msg, 2000);
+
+    // check for expired predictive tuning subscriptions and close those
+    CloseExpiredSubscriptions();
```

Comment:
Why move this here?

Reply:
Formerly, this check was done in DemuxRead - means, only while playing some channel. 

As we can not close subscriptions anymore when playback stops, we need to check even if we are not currently playing. Otherwise, the "last" subscription  would stay open until Kodi exits.